nixnet/types.py

```diff
@@ -230,3 +286,24 @@ def __repr__(self):
             self.echo,
             self.type,
             self.timestamp)
+
+
+class XnetFrame(FrameFactory):
+    """Create `Frame` based on `RawFrame` content."""
+
+    __slots__ = ()
+
+    @classmethod
+    def from_raw(cls, frame):
+        # type: (RawFrame) -> Frame
+        """Convert from RawFrame."""
+        frame_type = {
+            constants.FrameType.CAN_DATA: CanFrame,
+            constants.FrameType.CAN20_DATA: CanFrame,
+            constants.FrameType.CANFD_DATA: CanFrame,
+            constants.FrameType.CANFDBRS_DATA: CanFrame,
+            constants.FrameType.CAN_REMOTE: CanFrame,
+        }.get(frame.type)
+        if frame_type is None:
+            raise NotImplementedError("Unsupported frame type", frame.type)
+        return frame_type.from_raw(frame)
```

Comment:
@lilesj The purpose of this is for us to create the correct `Frame` object based on the type encoded in the data.  For example, more cases would be added for `LinFrame` and for each of the special frames.

It looked like all of these `FrameType` used the same data format.  Did I get this correct?